### ISSUE TITLE
var in lookup is undefined if task has no name in 1.8.2

### ISSUE THREAD

**Issue by MatthewHannigan** [ansible/ansible#10376]
If I define a var homevar: "HOME" and then have this action

```
- debug: msg="homevar is {{ lookup('env', homevar) }}"
```

I get a message:
`ERROR: Unexpected error in during lookup: 'homevar' is undefined`

However if I give the action a name

```
- name: lookup homevar
  debug: msg="homevar is {{ lookup('env', homevar) }}"
```

It works as expected, giving me the value of the environment variable HOME

Here's a complete playbook:

```

---
- hosts: localhost
  gather_facts: no
  vars:
    homevar:  "HOME"
  tasks:
    - name: show homevar
      debug: msg="homevar is {{ homevar }}"
    - name: lookup homevar
      debug: msg="homevar is {{ lookup('env', homevar) }}"
#   - name: lookup homevar without name
    - debug: msg="homevar is {{ lookup('env', homevar) }}"
```

**Comment by MatthewHannigan** [comment]
Looks similar to a few others so I made changes similar to [the fix for #5059](https://github.com/ansible/ansible/commit/2a976ac3137e3fb0a8cdbc68769b92e70bd460dc) to a copy of 1.8.2 and that seemed to fix it.  Upgrading to 1.8.4 also fixes it but upgrading is not an option on my main work machine right now.

**Comment by bcoca** [comment]
I'm going to close this as it is already solved in devel and 1.8.4.

@MatthewHannigan when we fix a problem in a minor release we push out another minor release. 1.8.3 and 1.8.4 only have security and bug fixes and should not really be considered an upgrade as much as a maintenance patch.

**Comment by MatthewHannigan** [comment]
Thanks @bcoca, understood and it's fine to close.
Part of the reason I raised this is that it seemed somewhat similar to other issues and they might have a common cause or fix.
